apps/sync/test/unit/adapters/bookmarks_test.js

```diff
@@ -0,0 +1,369 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+/* global
+  assert,
+  asyncStorage,
+  DataAdapters,
+  BOOKMARKS_COLLECTION_MTIME,
+  BOOKMARKS_SYNCTOID_PREFIX,
+  BookmarksHelper,
+  MockasyncStorage,
+  MockDatastore,
+  MockLazyLoader,
+  MockNavigatorDatastore,
+  require,
+  requireApp,
+  setup,
+  sinon,
+  suite,
+  suiteSetup,
+  suiteTeardown,
+  teardown,
+  test
+*/
+
+require('/apps/music/test/unit/mock_lazy_loader.js');
+require('/shared/test/unit/mocks/mock_navigator_datastore.js');
+require('/apps/system/test/unit/mock_asyncStorage.js');
+requireApp('sync/js/adapters/bookmarks.js');
+
+window.DataAdapters = {};
+
+suite('sync/adapters/bookmarks >', () => {
+  var realDatastore, realLazyLoader, realAsyncStorage, testCollectionData;
+  var updateBookmarksSpy;
+  var kintoCollection = {
+    list() {
+      return Promise.resolve({
+        data: testCollectionData
+      });
+    }
+  };
+
+  function getBookmarksStore() {
+    return navigator.getDataStores().then(stores => {
+      return Promise.resolve(stores[0]);
+    });
+  }
+
+  function verifyBookmarks(collectionItem, bookmarksItem) {
+    var payload = collectionItem.payload;
+    if (payload.type === 'bookmark' || payload.type === 'query') {
+      assert.equal(payload.bmkUri, bookmarksItem.url);
+      assert.equal(payload.title, bookmarksItem.name);
+    } else {
+      assert.equal(payload.type + ':' + payload.id, bookmarksItem.id);
```

Comment:
I guess this is not failing cause you are only testing `bookmark` type. Add tests for all the possible types, please.